### PR TITLE
API-397: Only view logged user jobs in “LAST OPERATIONS” widget

### PR DESCRIPTION
**Pull request by momoss**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/JobExecutionRepository.php [hunk 1]
```diff
@@ -56,25 +56,27 @@ public function createDatagridQueryBuilder()
...
      * Get data for the last operations widget
      *
-     * @param array $types Job types to show
+     * @param array       $types Job types to show
+     * @param string|null $user
      *
      * @return array
      */
-    public function getLastOperationsData(array $types)
+    public function getLastOperationsData(array $types, string $user = null)
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/JobExecutionRepository.php]: `?string $user = null`

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/JobExecutionRepository.php [hunk 2]
```diff
@@ -56,25 +56,27 @@ public function createDatagridQueryBuilder()
...
      * Get last operations query builder
      *
-     * @param array $types
+     * @param array       $types
+     * @param string|null $user
      *
      * @return \Doctrine\ORM\QueryBuilder
      */
-    protected function getLastOperationsQB(array $types)
+    protected function getLastOperationsQB(array $types, string $user = null)
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/JobExecutionRepository.php]: `?string $suser` (should not work without any user this way)

File: src/Pim/Bundle/ImportExportBundle/Manager/JobExecutionManager.php [hunk 3]
```diff
@@ -50,6 +61,9 @@ function ($type) {
             }
         );
 
-        return $this->repository->getLastOperationsData($types);
+        $token = $this->tokenStorage->getToken();
+        $user = null !==  $token ? $token->getUsername() : null;
```
Comment (skeleton) [inline src/Pim/Bundle/ImportExportBundle/Manager/JobExecutionManager.php]: $username

File: src/Pim/Bundle/ImportExportBundle/Manager/JobExecutionManager.php [hunk 4]
```diff
@@ -50,6 +61,9 @@ function ($type) {
             }
         );
 
-        return $this->repository->getLastOperationsData($types);
+        $token = $this->tokenStorage->getToken();
```
Comment (skeleton) [inline src/Pim/Bundle/ImportExportBundle/Manager/JobExecutionManager.php]: `if (null !== $this->tokenStorage) `

File: src/Pim/Bundle/EnrichBundle/tests/integration/Repository/JobExecutionRepositoryIntegration.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    /** @var JobLauncher */
+    protected $jobLauncher;
+
+    public function testGetLastOperationsData()
+    {
+        $this->jobLauncher->launchExport('csv_product_export', 'julia');
+        $this->jobLauncher->launchExport('csv_product_export', 'admin');
+
+        $result = $this->getJobExecutionRepository()->getLastOperationsData([], 'admin');
+        $result[0]['date'] = DateSanitizer::sanitize($result[0]['date']->format('c'));
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/tests/integration/Repository/JobExecutionRepositoryIntegration.php]: I would add a count of $result, to be sure only 1 jobs is returned

**Pull request merged**